src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -479,9 +479,27 @@ def show_service_principal(client, identifier):
     return client.get(object_id)
 
 
-def delete_service_principal(client, identifier):
-    object_id = _resolve_service_principal(client, identifier)
-    client.delete(object_id)
+def delete_service_principal(identifier):
```

Comment:
I thought about that.  But after you delete a service principal, you will not be able to get it back with the same object id, hence the existing role assignments will become invalid which we should just clean up.